File: snippets/cpp/mst/kruskal.cpp
ll kruskal(int n, vector<array<int,3>> &edges) {
    sort(all(edges));
    DSU dsu(n); ll cost = 0;
    for (auto [w, u, v] : edges)
        if (dsu.unite(u, v)) cost += w;
    return cost;
}

File: snippets/rust/mst/kruskal.rs
// Requires `DSU` from the `dsu` snippet. Edges are (weight, u, v).
fn kruskal(n: usize, mut edges: Vec<(i64, usize, usize)>) -> i64 {
    edges.sort();
    let mut dsu = DSU::new(n);
    let mut cost = 0i64;
    for (w, u, v) in edges {
        if dsu.unite(u, v) { cost += w; }
    }
    cost
}
